api/src/main/java/org/openmrs/module/registrationcore/api/impl/RegistrationCoreProperties.java

```diff
@@ -29,9 +29,14 @@ public void setApplicationContext(ApplicationContext applicationContext) {
         this.applicationContext = applicationContext;
     }
 
-    public Integer getOpenMrsIdentifierSourceId() {
+	public Integer getOpenMrsIdentifierSourceId() {
         String propertyName = RegistrationCoreConstants.GP_OPENMRS_IDENTIFIER_SOURCE_ID;
-        return getIntegerProperty(propertyName);
+		return getIntegerProperty(propertyName);
+	}
+	
+	public String getOpenMrsIdentifierSourceUuid() {
+		String propertyName = RegistrationCoreConstants.GP_OPENMRS_IDENTIFIER_SOURCE_ID;
+        return getProperty(propertyName);
```

Comment:
I find this a little confusing.  If both of these methods exist, I'd prefer to make them both work correctly, regardless of whether an id or uuid is in the GP.  So, in the getOpenMRSIdentifierSourceId method, if it is an integer, return it.  Otherwise, get the source by uuid, and then return the id of the source.  Likewise, in the getOpenMrsIdentifierSourceUuid method, if the property is not an Integer, just return it.  Otherwise, get the source by id, and return the uuid of the source.

Reply:
Deprecated and kept `getOpenMrsIdentifierSourceId()` for backwards compatibility. We should use `getOpenMrsIdentifierSource()`, hope that clears the confusion

Reply:
Still not sure about this naming.  This doesn't return an identifier source, it returns a String global property value representing either the id or uuid of the source.

Reply:
Ahh, sure.. Nice catch. Let me fix the naming

Reply:
@mseaton  corrected :)